File: add.py
# multiplication.py

from http.server import BaseHTTPRequestHandler, HTTPServer
from urllib.parse import urlparse, parse_qs

# Classe de gestion des requêtes HTTP
class RequestHandler(BaseHTTPRequestHandler):
    def do_GET(self):
        # Analyse de l'URL
        parsed_path = urlparse(self.path)
        query_params = parse_qs(parsed_path.query)

        # Vérification des paramètres requis
        if 'number1' in query_params and 'number2' in query_params:
            # Récupération des nombres et calcul de la multiplication
            number1 = int(query_params['number1'][0])
            number2 = int(query_params['number2'][0])
            result = number1 * number2

            # Envoi de la réponse HTTP
            self.send_response(200)
            self.send_header('Content-type', 'text/html')
            self.end_headers()

            # Corps de la réponse
            self.wfile.write(bytes("<html><head><title>Résultat de la multiplication</title></head>", "utf-8"))
            self.wfile.write(bytes("<body>", "utf-8"))
            self.wfile.write(bytes("<h2>Résultat de la multiplication en Python</h2>", "utf-8"))
            self.wfile.write(bytes(f"<p>Résultat de {number1} * {number2} : {result}</p>", "utf-8"))
            self.wfile.write(bytes("</body></html>", "utf-8"))
        else:
            # En cas de paramètres manquants, renvoyer une erreur 400 (Bad Request)
            self.send_error(400, 'Bad Request')

# Fonction principale pour démarrer le serveur
def run(server_class=HTTPServer, handler_class=RequestHandler, port=8080):
    server_address = ('', port)
    httpd = server_class(server_address, handler_class)
    print(f'Starting server on port {port}...')
    httpd.serve_forever()

if __name__ == '__main__':
    run()

File: add.php
<?php
// Vérification si la méthode de requête est POST
if ($_SERVER["REQUEST_METHOD"] == "POST") {
    // Récupération des nombres depuis le formulaire
    $number1 = $_POST["number1"];
    $number2 = $_POST["number2"];

    // Vérification si les nombres sont des entiers
    if (is_numeric($number1) && is_numeric($number2)) {
        // Calcul de la multiplication
        $result = $number1 * $number2;
        // Affichage du résultat
        echo "Résultat de la multiplication : $result";
    } else {
        // Si les nombres ne sont pas valides, affichage d'un message d'erreur
        echo "Les nombres saisis ne sont pas valides.";
    }
}

echo "<!DOCTYPE html>

<html lang=\"en\">\n
    <head>\n
    <meta charset=\"UTF-8\">\n
    <meta http-equiv=\"X-UA-Compatible\" content=\"IE=edge\">\n
    <title>Basic Calculator</title>\n
</head>\n
<body>";
echo "<h1>Result :</h1>";
echo "<p>$a x $b = $result</p>";
echo "</body>\n
</html>\n";
?>